### DISCUSSION TITLE
Sybase driver

### DISCUSSION THREAD

**Discussion by EvgenyKiselgof** [apache/superset#18290] [Q&A / Help]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Be able to connect to Sybase DB, like ASE and IQ

**Describe alternatives you've considered**
Support of FreeTDS with pyodbc

**Additional context**
I already saw few topics about this issue, but no one describes exactly how the resolve it and how it can be integrated.

**Comment by srinify** [comment]
Hi @EvgenyKiselgof do you know if there's an up-to-date SQLAlchemy + DB-API 2.0 driver for Sybase? I wrote up what's needed to add new database support in Superset here (https://preset.io/blog/building-database-connector/)

This page seems to indicate that the sqlalchemy dialect is deprecated? 

https://docs.sqlalchemy.org/en/14/dialects/sybase.html#module-sqlalchemy.dialects.sybase.pysybase

**Comment by EvgenyKiselgof** [comment]
Hi,
Thanks for you response :)
In alchemy documentation there is 3 or 4 different drivers to use.
I know for sure that pyodbc is working well with FreeTDS odbc driver,
currsntly im using just python script  to query Sybase DB.
Maybe i don't understand well the documentation about the database support.
Maybe you can provide me some example file.



בתאריך יום ב׳, 26 באפר׳ 2021, 17:41, מאת Srini Kadamati ‏<
***@***.***>:

**Comment by srinify** [comment]
@EvgenyKiselgof I recently blogged about this actually:

https://preset.io/blog/building-database-connector/

I can follow up in a week or 2 with some research into Sybase! Do you have Sybase running anywhere? One of the challenges is testing this, I don't have a Sybase database anywhere!

**Comment by EvgenyKiselgof** [comment]
Hi,
I saw the blog, but it not so clear to me :(  I just learning.
but what I did for now is, simple python script, that uses pyodbc and
freeTDS odbc driver,  the script is connecting to the Sybase DB running my
queries and then Insert the results CSV files to PostgresDB and the with
Superset I'm reading the data actually from Postgres :)
I believe that if I can understand well the  db_engine_specs construction I
will be able to create something :)
I have Sybase DB but I can't share it.


On Thu, Apr 29, 2021 at 5:28 PM Srini Kadamati ***@***.***>
wrote:

> @EvgenyKiselgof <https://github.com/EvgenyKiselgof> I recently blogged
> about this actually:
>
> https://preset.io/blog/building-database-connector/
>
> I can follow up in a week or 2 with some research into Sybase! Do you have
> Sybase running anywhere? One of the challenges is testing this, I don't
> have a Sybase database anywhere!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/superset/issues/14337#issuecomment-829284627>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANMN2AIDYNF2GR7BDJSEXNLTLFUHDANCNFSM43RJGC3Q>
> .
>


-- 
Thanks,
Evgeny Kiselgof

**Comment by srinify** [comment]
Hmm @EvgenyKiselgof do you know others who are interested in Sybase <> Superset? I'd be happy to bring everyone into the 'same room' and help make this happen.

**Reply by ricardodinix** [reply]
I am very interested in the integration between Sybase IQ and Apache Superset. I'm joined by at least 5 other professionals interested in the subject.

**Comment by EvgenyKiselgof** [comment]
Hi,
Do we able somehow to disable the Limit suffix in queries?
Cause i found that ase didnt support LIIMIT keyword, and i didnt find any
way to enable this Keyword.
So i belive if i can somehow to disable the row limit from queries it will
work.
Currsnrly im able to connect to sybase and read the schema.

בתאריך יום ה׳, 29 באפר׳ 2021, 17:57, מאת Srini Kadamati ‏<
***@***.***>:
